os/drivers/video/video_null_driver.c

```diff
@@ -0,0 +1,1554 @@
+	uint16_t type;				/* enum #v4l2_ctrl_type */
+	char name[32];				/* Name of control */
+	int32_t min;				/* Minimum value */
+	int32_t max;				/* Maximum value */
+	uint32_t step;				/* step */
+	int32_t default_value;		/* Default value */
+	int32_t current_value;		/* Current value */
+};
+
+struct v4l2_interval_s {
+	uint32_t min_numerator;		/* numerator */
+	uint32_t max_numerator;		/* numerator */
+	uint32_t step_numerator;	/* numerator */
+};
+
+struct v4l2_frames_s {
+	uint32_t wWidth;
+	uint32_t wHeight;
+	uint32_t type;
+	struct v4l2_interval_s intervals[MAX_INTERVAL];
+	uint32_t intervalcount;
+};
+
+struct v4l2_format_s {
+	char description[V4L2_FMT_DSC_MAX];	/* Description string */
+	uint16_t type;				/* v4l2_buf_type */
+	uint32_t pixelformat;		/* Format fourcc      */
+	uint32_t colorspace;		/* Color space */
+	uint8_t bpp;				/* Bits per pixel */
+	struct v4l2_frames_s frames[MAX_FRAMES];	/* Pointer to frames object */
+	uint16_t framecount;		/* Number of frame supported */
+};
+
+struct dummy_null_priv_s {
+	int16_t crefs;				/* Reference count on the driver instance */
+	sem_t sem;					/* Used to maintain mutual exclusive access */
+	FAR void *priv_data;		/* Video private data */
+	struct v4l2_mapping_s *ctrl;	/* Controls supported by this device */
+	uint16_t ctrlcount;			/* Number of control supported by this device */
+	struct v4l2_format_s *fmt;	/* Supported formats */
+	uint16_t fmtcount;			/* Number of formats supported */
+	struct work_s trfwork;		/* Work queue handle */
+	enum v4l2_buf_type reqtype;	/* Type of the buffer stream or capture */
+	uint32_t *reqbuff;			/* User request buffer */
+	uint32_t reqsize;			/* User request buffer size */
+	uint32_t *trfbuff;			/* Transfer buffer */
+	uint32_t trfsize;			/* Data recevied so far from camera */
+	bool reqcancel;				/* Request status */
+	struct v4l2_format_s *def_format;	/* Reference to default format */
+	struct v4l2_format_s *cur_format;	/* Reference to current format */
+	struct v4l2_frames_s *cur_frame;	/* Reference to current frame */
+};
+typedef struct dummy_null_priv_s null_priv_t;
+
+struct video_devops_s dummy_null_video_ops = {
+	.open = video_null_open,
+	.close = video_null_close,
+	.do_halfpush = video_null_do_halfpush,
+	.set_buftype = video_null_set_buftype,
+	.set_buf = video_null_set_buf,
+	.cancel_dma = video_null_cancel_dma,
+	.get_range_of_fmt = video_null_get_range_of_fmt,
+	.get_range_of_framesize = video_null_get_range_of_framesize,
+	.try_format = video_null_try_format,
+	.set_format = video_null_set_format,
+	.get_range_of_frameinterval = video_null_get_range_of_frameinterval,
+	.set_frameinterval = video_null_set_frameinterval,
+	.get_range_of_ctrlvalue = video_null_get_range_of_ctrlvalue,
+	.get_menu_of_ctrlvalue = video_null_get_menu_of_ctrlvalue,
+	.get_ctrlvalue = video_null_get_ctrlvalue,
+	.set_ctrlvalue = video_null_set_ctrlvalue,
+	.refresh = video_null_refresh,
+};
+
+struct v4l2_format_s supported_formats[] = {
```

Comment:
we'd better provide kconfig of null driver to get supported format list...
in menuconfig, we should display all of format list, then user can select some of those to test it.
How do you think?

Reply:
ok we will check for the feasibility and make separate patch